packages/block-editor/src/components/grid-interactivity/grid-visualizer.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __experimentalUseDropZone as useDropZone } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { __unstableUseBlockElement as useBlockElement } from '../block-list/use-block-props/use-block-refs';
+import BlockPopoverCover from '../block-popover/cover';
+import { getComputedCSS } from './utils';
+import { parseDropEvent } from '../use-on-block-drop';
+import { store as blockEditorStore } from '../../store';
+
+export function GridVisualizer( { clientId } ) {
+	const blockElement = useBlockElement( clientId );
+	if ( ! blockElement ) {
+		return null;
+	}
+	return (
+		<GridVisualizerGrid
+			clientId={ clientId }
+			blockElement={ blockElement }
+		/>
+	);
+}
+
+function getGridInfo( blockElement ) {
+	const gridTemplateColumns = getComputedCSS(
+		blockElement,
+		'grid-template-columns'
+	);
+	const gridTemplateRows = getComputedCSS(
+		blockElement,
+		'grid-template-rows'
+	);
+	const numColumns = gridTemplateColumns.split( ' ' ).length;
+	const numRows = gridTemplateRows.split( ' ' ).length;
+	const numItems = numColumns * numRows;
+	return {
+		numColumns,
+		numRows,
+		numItems,
+		style: {
+			gridTemplateColumns,
+			gridTemplateRows,
+			gap: getComputedCSS( blockElement, 'gap' ),
+			padding: getComputedCSS( blockElement, 'padding' ),
+		},
+	};
+}
+
+function range( start, length ) {
+	return Array.from( { length }, ( _, i ) => start + i );
+}
+
+function GridVisualizerGrid( { clientId, blockElement } ) {
+	const [ gridInfo, setGridInfo ] = useState( () =>
+		getGridInfo( blockElement )
+	);
+	const [ hasCellPointerEvents, setHasCellPointerEvents ] = useState( false );
+
+	useEffect( () => {
+		const observers = [];
+		for ( const element of [ blockElement, ...blockElement.children ] ) {
+			const observer = new window.ResizeObserver( () => {
+				setGridInfo( getGridInfo( blockElement ) );
+			} );
+			observer.observe( element );
+			observers.push( observer );
+		}
+		return () => {
+			for ( const observer of observers ) {
+				observer.disconnect();
+			}
+		};
+	}, [ blockElement ] );
+
+	useEffect( () => {
+		function onGlobalDrag() {
+			setHasCellPointerEvents( true );
+		}
+		function onGlobalDragEnd() {
+			setHasCellPointerEvents( false );
+		}
+		document.addEventListener( 'drag', onGlobalDrag );
+		document.addEventListener( 'dragend', onGlobalDragEnd );
+		return () => {
+			document.removeEventListener( 'drag', onGlobalDrag );
+			document.removeEventListener( 'dragend', onGlobalDragEnd );
+		};
+	}, [] );
+
+	return (
+		<BlockPopoverCover
+			className={ classnames( 'block-editor-grid-visualizer', {
```

Comment:
Somewhere a little `z-index` jig in mobile view is required as it sits over the sidebar

https://github.com/WordPress/gutenberg/assets/6458278/a68b85ee-1889-4a10-be6b-9095cbf11b07

Reply:
Hm. Why is this happening? When I look in the inspector the visualiser has `z-index = 30` and the list view has `z-index = 10000` 😕

Reply:
I suspect they're in different stacking contexts? Or a new one needs to be created around the grid visualizer. 

This tool is pretty handy to work it out: https://github.com/gwwar/z-context

Reply:
Yeah they're in different stacking contexts but what's causing the issue is the popover content moving to the bottom of the DOM when the viewport is smaller than 782px:

<img width="1848" alt="Screenshot 2024-03-18 at 5 07 37 PM" src="https://github.com/WordPress/gutenberg/assets/8096000/eb8b53a0-af06-4478-991e-8674dd339802">

I'm not sure what the role of `components-popover__fallback-container` is in popover logic, but it appears after the sidebar and is a sibling of `#wpwrap` so I suspect the only way to fix this with CSS would be to assign a non-negative z-index to `#wpwrap`. I doubt that would be free of consequences though 😅